src/main/java/com/tfgm/typehandler/JSONMapLongHandler.java

```diff
@@ -0,0 +1,91 @@
+package com.tfgm.typehandler;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.io.Reader;
+import java.sql.CallableStatement;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Map;
+import org.apache.ibatis.type.*;
+import org.postgresql.util.PGobject;
+
+/** Maps a map to a flat postgres JSONB field. */
+@SuppressWarnings("DuplicatedCode")
+@MappedJdbcTypes(JdbcType.OTHER)
+@MappedTypes(Map.class)
+@Alias(value = "JSONMapLongHandler")
+public class JSONMapLongHandler extends BaseTypeHandler<Map<String, Long>> {
```

Comment:
When using the Map<String,Object> handler the bigint from the db was being treated as an int instead of a Long causing issues as it still managed to save it into the variable correctly for some reason